### PR TITLE
Fix function pointer representation

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@amcasey, @kevinh-ms

**Comment by KevinH-MS** [comment]
@cston could you clarify why the old code didn't work?  It seemed a lot simpler/clearer.

**Comment by cston** [comment]
Previously the tests failed on 64-bit because the values used as placeholders for function pointers were ints rather than pointers. The tests now create actual pointer instances. Other than that, the tests should be equivalent to the previous code.

**Comment by KevinH-MS** [comment]
Gotcha

File: src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs [hunk 1]
```diff
@@ -12,40 +14,71 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
...
+        this.pfn = (int*)p;
+    }
+    int* pfn;
 }";
-            const int ptr = 0x0;
-            GetMemberValueDelegate getMemberValue = (v, m) => (m == "pfn") ? CreateDkmClrValue(ptr, type: new DkmClrType(FunctionPointerType.Instance)) : null;
-            var runtime = new DkmClrRuntimeInstance(ReflectionUtilities.GetMscorlibAndSystemCore(GetAssembly(source)), getMemberValue: getMemberValue);
-            using (runtime.Load())
+            var assembly = GetUnsafeAssembly(source);
+            unsafe
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs]: ```unsafe``` unnecessary?

**Comment by KevinH-MS** [comment]
LGTM

File: src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs (hunk 1 above)
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Test/ResultProvider/FunctionPointerTests.cs]: Fixed, thanks.

**Pull request merged**